lib/ukstore/include/uk/store.h

```diff
@@ -29,3 +29,350 @@
+		.get.e_type = (e_get),					\
+		.set.e_type = (e_set),					\
+		.flags      = UK_STORE_ENTRY_FLAG_STATIC,		\
+		.cookie     = NULL					\
+	}
+
+/* Do not call directly */
+#define _UK_STORE_STATIC_ENTRY(entry, lib_str, e_type, e_get, e_set)	\
+	__UK_STORE_STATIC_ENTRY(entry, lib_str, e_type, e_get, e_set)
+
+/**
+ * Adds an entry to the entry section of a library.
+ *
+ * @param entry the entry in the section
+ * @param e_type type for the entry, e.g., s8, u16, charp
+ * @param e_get getter pointer (optional, can be NULL)
+ * @param e_set setter pointer (optional, can be NULL)
+ */
+#define UK_STORE_STATIC_ENTRY(entry, e_type, e_get, e_set)	\
+	_UK_STORE_STATIC_ENTRY(entry, STRINGIFY(__LIBNAME__),	\
+		e_type, e_get, e_set)
+
+const struct uk_store_entry *
+_uk_store_get_static_entry(unsigned int libid, const char *e_name);
+
+static inline const struct uk_store_entry *
+_uk_store_get_entry(unsigned int libid, const char *f_name __unused,
+			const char *e_name)
+{
+	return _uk_store_get_static_entry(libid, e_name);
+}
+
+/**
+ * Searches for an entry in a folder in a library. Increases the refcount.
+ *
+ * @param libname the name of the library to search in
+ * @param foldername the name of the folder to search (NULL for static entries)
+ * @param entryname the name of the entry to search for
+ * @return the found entry or NULL
+ */
+#define uk_store_get_entry(libname, foldername, entryname)	\
+	_uk_store_get_entry(uk_store_libid(libname), foldername, entryname)
+
+/**
+ * Decreases the refcount. When it reaches 0, the memory is freed
+ *
+ * Guard this with `#if CONFIG_LIBUKSTORE`
+ *
+ * @param entry the entry to release
+ */
+#define uk_store_release_entry(entry)	\
+	_uk_store_release_entry((entry))
+
+void
+_uk_store_release_entry(const struct uk_store_entry **entry);
+
+int
+_uk_store_get_u8(const struct uk_store_entry *e, __u8 *out);
+int
+_uk_store_get_s8(const struct uk_store_entry *e, __s8 *out);
+int
+_uk_store_get_u16(const struct uk_store_entry *e, __u16 *out);
+int
+_uk_store_get_s16(const struct uk_store_entry *e, __s16 *out);
+int
+_uk_store_get_u32(const struct uk_store_entry *e, __u32 *out);
+int
+_uk_store_get_s32(const struct uk_store_entry *e, __s32 *out);
+int
+_uk_store_get_u64(const struct uk_store_entry *e, __u64 *out);
+int
+_uk_store_get_s64(const struct uk_store_entry *e, __s64 *out);
+int
+_uk_store_get_uptr(const struct uk_store_entry *e, __uptr *out);
+int
+_uk_store_get_charp(const struct uk_store_entry *e, char **out);
+int
+_uk_store_get_ncharp(const struct uk_store_entry *e, char **out,
+	size_t maxlen);
+
+/**
+ * Calls the getter to get it's value.
+ * The caller is responsible for freeing the returned value if it is a charp.
+ *
+ * @param entry the entry to call the getter from
+ * @param outtype the type of the variable used (e.g., u8, s8, u16, uptr, ...)
+ * @param out space to store the result of the call
+ * @return 0 or error on fail
+ */
+#define uk_store_get_value(entry, outtype, out)	\
```

Comment:
This may fail when you specify ncharp as type because of the missing maxlen argument. I would actually exclude ncharp from the macro resolution. See my comment below....